src/umpire/resource/HostSharedMemoryResourceFactory.cpp

```diff
@@ -0,0 +1,60 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#include "umpire/resource/HostSharedMemoryResourceFactory.hpp"
+
+#include "umpire/resource/HostSharedMemoryResource.hpp"
+#include "umpire/util/Macros.hpp"
+#include "umpire/util/make_unique.hpp"
+
+namespace umpire {
+namespace resource {
+
+bool HostSharedMemoryResourceFactory::isValidMemoryResourceFor(
+    const std::string& name) noexcept
+{
+  if (name.find("SHARED") != std::string::npos) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+std::unique_ptr<resource::MemoryResource> HostSharedMemoryResourceFactory::create(
+    const std::string& name, int id)
+{
+  return create(name, id, getDefaultTraits());
+}
+
+std::unique_ptr<resource::MemoryResource> HostSharedMemoryResourceFactory::create(
+    const std::string& name, int id, MemoryResourceTraits traits)
+{
+  if (traits.scope != MemoryResourceTraits::shared_scope::node) {
+    UMPIRE_ERROR("HostSharedMemoryResource only supports shared_scope::node");
+  }
+  return util::make_unique<HostSharedMemoryResource>(Platform::undefined, name, id,
```

Comment:
In this instance, and in general, I am confused by the use of undefined as a platform type (like in the traits file too). What is happening with this?

Reply:
What should the platform be?

Reply:
It probably should be Platform::host.

Reply:
Host.

Reply:
Done!  Nice catch!

Reply:
But! ... It's not done lol what about the isAccessibleFrom function? Hold on, I'll link to it...